### DISCUSSION TITLE
KeyError: 3834 in YOLOv5 Detection Script

### DISCUSSION THREAD

**Discussion by Harshit28j** [ultralytics/yolov5#12186] [Q&A] [answered]
I am encountering a KeyError: 3834 in my YOLOv5 detection script and would appreciate some guidance on resolving this issue. I have created a custom dataset to test and understand YOLO. These are the names and classes specified in my custom_file.yaml

**CLI:**
```
python detect.py --weights C:\Users\Harsh\runs\detect\train15\weights\last.pt --source C:\Users\Harsh\OneDrive\Desktop\v.mp4
```
**Error Message:**
```
File "C:\Users\Harsh\OneDrive\Documents\Projects\custom detection tool\yolov5-master\detect.py", line 175, in run
    s += f"{n} {names[int(c)]}{'s' * (n > 1)}, "  # add to string
KeyError: 3834
```
**Yaml File:**

![image](https://github.com/ultralytics/yolov5/assets/48647625/89632a77-ccd2-4f06-aba3-61ca2badd031)

Any insights on resolving the KeyError: 3834 would be highly appreciated.

**Comment by inesxrodrigues** [comment]
Hello,

did you train the model with YOLOv5 also? Because I got the exact same error when I tried to run detect.py of YOLOv5 using a model I trained with YOLOv8.

**Reply by Harshit28j** [reply, marked as answer]
Yes @inesxrodrigues , I figured it out, my friend. This issue is occurring because the class names I have given in my custom_file.yaml are not equal to the number of classes in my dataset. I suggest you organize your dataset properly, ensuring that class names are correct, and specify the correct number of classes.
